framework/test/jest/specs/unit/controller/application_state.spec.js

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+const ApplicationState = require('../../../../../src/controller/application_state');
+
+jest.mock('os', () => ({
+	platform: jest.fn(() => 'platform'),
+	release: jest.fn(() => 'release'),
+}));
+
+describe('Application State', () => {
+	let applicationState = null;
+	const initialState = {
+		version: '1.0.0-beta.3',
+		wsPort: '3001',
+		httpPort: '3000',
+		minVersion: '1.0.0-beta.0',
+		protocolVersion: '1.0',
+		nethash: 'test broadhash',
+		nonce: 'test nonce',
+	};
+	const mockedState = {
+		os: 'platformrelease',
+		version: '1.0.0-beta.3',
+		wsPort: '3001',
+		httpPort: '3000',
+		minVersion: '1.0.0-beta.0',
+		protocolVersion: '1.0',
+		nethash: 'test broadhash',
+		broadhash: 'test broadhash',
+		height: 1,
+		nonce: 'test nonce',
+	};
+	const logger = {
+		debug: jest.fn(),
+		error: jest.fn(),
+	};
+
+	const channel = {
+		publish: jest.fn(),
+	};
+
+	beforeEach(() => {
+		// Act
+		applicationState = new ApplicationState({
+			initialState,
+			logger,
+		});
+	});
+
+	describe('#constructor', () => {
+		it('should initiate the application state', () => {
+			// Assert
+			expect(applicationState.logger).toBe(logger);
+			expect(applicationState.stateChannel).toBe(undefined);
+			expect(applicationState.state).toEqual(mockedState);
+		});
+	});
+
+	describe('#get state', () => {
+		it('should call get state', () => {
+			// Arrange
+			const spy = jest.spyOn(applicationState, 'state', 'get');
```

Comment:
no need to spy imo.
As long as `state` is equaled to mocked state, test should be sufficient